Tests/SwiftSyntaxTest/SyntaxConvenienceMethodsTests.swift

```diff
@@ -2,14 +2,16 @@ import XCTest
 import SwiftSyntax
 
 public class SyntaxConvenienceMethodsTests: XCTestCase {
-
-  public func testFloatingValues() {
-    testFloatingValue(text: "5.3_8", expectedValue: 5.38)
-    testFloatingValue(text: "12e3", expectedValue: 12000.0)
-    testFloatingValue(text: "32E1", expectedValue: 320.0)
-    testFloatingValue(text: "0xdEFACE.C0FFEEp+1", expectedValue: 0xdEFACE.C0FFEEp+1)
-    testFloatingValue(text: "0xaffab1e.e1fP-2", expectedValue: 0xaffab1e.e1fP-2)
-    testFloatingValue(text: "🥥", expectedValue: nil)
+  public func testFloatValues() {
+    testFloatValue(text: "5.3_8", expectedValue: 5.38)
+    testFloatValue(text: "12e3", expectedValue: 12000.0)
+    testFloatValue(text: "32E1", expectedValue: 320.0)
+    testFloatValue(text: "0xdEFACE.C0FFEEp+1", expectedValue: 0xdEFACE.C0FFEEp+1)
+    testFloatValue(text: "0xaffab1e.e1fP-2", expectedValue: 0xaffab1e.e1fP-2)
+    testFloatValue(text: "🥥", expectedValue: nil)
+    // The following results in a valid FloatLiteralExprSyntax, but because Double.greatestFiniteMagnitude
+    // is 1.7976931348623157e+308, floatLiteralValue should be nil.
```

Comment:
It should _not_ be `nil`. If you write `let x = 1.7976931348623157e+309`, you get an actual `Double` which is `infinity`. This is one of those divergences I discuss above between the string conversion APIs and the float literal APIs.

Reply:
The last commit was about removing `must_uphold_invariant` and verifying that unit tests still pass. I have not yet explored additional changes.